src/pool/srv_pool.c

```diff
@@ -2338,6 +2347,31 @@ ds_pool_connect_handler(crt_rpc_t *rpc)
 	}
 	D_ASSERT(prop != NULL);
 
+	global_ver_entry = daos_prop_entry_get(prop, DAOS_PROP_PO_GLOBAL_VERSION);
+	D_ASSERT(global_ver_entry != NULL);
+	global_ver = global_ver_entry->dpe_val;
+	/*
+	 * Reject pool connection if client is unable to access newer pool.
+	 */
+	if (handler_version >= 5) {
+		struct pool_connect_v5_in *in1 = (struct pool_connect_v5_in *)in;
+
+		if (in1->pci_pool_version <= DAOS_POOL_GLOBAL_VERSION) {
+			if (global_ver > pool_max_supported_version[in1->pci_pool_version]) {
```

Comment:
We should be checking `in1->pci_pool_version` against the length of the array before indexing into it.

Reply:
oh, see in1->pci_pool_version <= DAOS_POOL_GLOBAL_VERSION, this is length check? maybe i should add a assertion that pool_max_supported_version length is DAOS_POOL_GLOBAL_VERSION

Reply:
maybe it is ok, as pool_max_supported_version is defined using macro.
static unsigned int pool_max_supported_version[DAOS_POOL_GLOBAL_VERSION + 1] = {
        1,
        1,
};

Reply:
Ah, thanks, I missed that. I wonder if it would be valuable to have a separate NUM_POOL_VERSIONS macro. Will it always be global version + 1? Or will that change once we begin bumping the version? Will the array actually grow?

Reply:
This is done in the latest codes